### ISSUE TITLE
IPv6 Gateway not allowed to be outside subnet

### ISSUE THREAD

**Issue by DennisGlindhart** [containernetworking/plugins#233]
When using the following configuration where the gateway is outside of the subnet the following error is received:

**invalid range set 0: gateway 2001:db8:5::1 not in network 2001:db8:5:333::/64**

```
    "ipam": {
        "type": "host-local",
        "subnet": "2001:db8:5:333::/64",
        "gateway": "2001:db8:5::1",
    }
```

It's not allowed in IPv4, but in IPv6 this is perfectly ok to have the gateway in a different subnet as long as the gateway is on the same L2 link. (Gateway Link-Local is looked up via Neighbor Discovery).

This is AFAIK fairly normal for ISP's to assign i.e. /48 or /56 subnet and only provide a single gateway address.

A workaround is to change the subnet size and use range to ensure that IP's are in the intended range.

```
    "ipam": {
        "type": "host-local",
        "subnet": "2001:db8:5:333::/48",
        "gateway": "2001:db8:5::1",
        "rangeStart": "2001:db8:5:333::",
        "rangeEnd": "2001:db8:5:333:ffff:ffff:ffff:ffff",
    }
```

but this results in an address in the container/endpoint with the /48 prefix instead of /64 which is preferable for IPv6 to work properly.
```
# ip a
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UNKNOWN group default 
    link/ether xx:xx:xx:xx:xx:xx brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet6 2001:db8:5:333::5/48 scope global 
       valid_lft forever preferred_lft forever
    inet6 fe80::x:x/64 scope link 
       valid_lft forever preferred_lft forever
```

**Comment by squeed** [comment]
Oh duh, good catch.

**Comment by squeed** [comment]
Fixed in #236, closing.
